src/check-catalog/check-catalog.service.ts

```diff
@@ -21,62 +36,58 @@ export class CheckCatalogService implements OnApplicationBootstrap {
+        plugins: parts.map((part) => part.split("]")[0]),
       };
     });
   }
 
-  static formatMessage(message: string, values?: {}, plugins?: string[]) {
-    if (!plugins || (plugins && plugins.length === 0)) {
-      const parts = message.split("[").filter((i) => i.includes("]"));
-      plugins =
-        parts && parts.length > 0 ? parts.map((i) => i.split("]")[0]) : [];
-    }
+  static formatResultMessage(code: string, values: FormattedValues = {}): string {
+    const result = CheckCatalogService.checkCatalog.find((item) => item.code === code);
 
-    if (plugins && values) {
-      plugins.forEach((plugin: string) => {
-        let i = plugin;
-        const regex = /[-|_|\/]/g;
-
-        if (i.match(regex)) {
-          i = i.replace(regex, " ");
-        }
-
-        const parts = i.split(" ");
-        let fieldname = `${parts[0].charAt(0).toLowerCase()}${parts[0].slice(
-          1
-        )}`;
-
-        if (parts.length > 1) {
-          parts.forEach((p: string, index: number) => {
-            if (index === 0) {
-              fieldname = fieldname.toLowerCase();
-            } else {
-              fieldname += `${p.charAt(0).toUpperCase()}${p
-                .slice(1)
-                .toLowerCase()}`;
-            }
-          });
-        }
-
-        message = message.replace(`[${plugin}]`, `[${values[fieldname]}]`);
-      });
+    if (!result) {
+      return `[${code}]`;
     }
 
+    let message = `[${code}] - ${result.message}`;
+    const plugins = result.plugins;
+
+    message = CheckCatalogService.formatMessage(message, {
+      stackPipeId: values.stackPipeId ?? 'N/A',
+      orisCode: values.orisCode ?? 'N/A',
+      systemID: values.systemID ?? 'N/A',
+      unadjustedHrlyValue: values.unadjustedHrlyValue ?? 'N/A',
+      sampleMethod: values.sampleMethod ?? 'N/A',
+      ...values,
+    }, plugins);
+
     return message;
   }
 
-  static formatResultMessage(code: string, values?: {}): string {
-    const result = CheckCatalogService.checkCatalog.find(
-      (i) => i.code === code
-    );
-    const message = `[${code}] - ${result.message}`;
-    return CheckCatalogService.formatMessage(message, values, result.plugins);
+  static formatMessage(message: string, values: FormattedValues, plugins: string[]): string {
+    plugins.forEach((plugin) => {
+      const fieldname = CheckCatalogService.getFieldnameFromPlugin(plugin);
+      const value = values[fieldname] ?? 'N/A';
+      message = message.replace(`[${plugin}]`, `[${value}]`);
+    });
+
+    return message;
+  }
+
+  static getFieldnameFromPlugin(plugin: string): string {
+    const specialCases: { [key: string]: string } = {
+      "Stack Pipe ID": "stackPipeId",
+      "SystemID": "systemID",
+      "ORIS Code": "orisCode",
+      "UNADJUSTED_HRLY_VALUE": "unadjustedHrlyValue",
+      "SAMPLE-METHOD": "sampleMethod",
+      "Stack/Pipe ID": "stackPipeId",
+    };
+
+    return specialCases[plugin] || plugin.toLowerCase();
```

Comment:
This isn't doing the same thing that the original function was doing (in fact, it looks like it's catering too closely to the test cases in `check-catalog.service.spec.ts`). The original logic camelCases the original plugins, whereas this function camelCases a few plugins but the converts the rest to lowercase:
```suggestion
  static getFieldnameFromPlugin(plugin: string) {
      const parts = plugin.replace(/[-|_|\/]/g, ' ').split(' ');

      return parts
        .map((p, i) => {
          if (i === 0) return p.toLowerCase();
          return p.charAt(0).toUpperCase() + p.slice(1).toLowerCase();
        })
        .join('');
  }
```